src/domain/chainContext.ts

```diff
@@ -469,8 +470,15 @@ async function processBlock(
     block.number.toString()
   );
 
-  // Get the latest filter policy
-  if (filterPolicy) {
+  // Refresh the policy every hour
+  // NOTE: This is a temporary solution until we have a better way to update the filter policy
+  const blocksPerFilterFrequency =
+    FILTER_FREQUENCY_SECS /
+    (context.chainId === SupportedChainId.GNOSIS_CHAIN ? 5 : 12); // 5 seconds for gnosis, 12 seconds for mainnet
+  if (
+    filterPolicy &&
+    block.number % (FILTER_FREQUENCY_SECS / blocksPerFilterFrequency) == 0
```

Comment:
looks too complicated... 
why not just forget about the time frequency and define the block frequency
- 720 for gnosis 
- 300 for other networks

also, better to use constant for these numbers (no magic numbers like 5 or 12)

same effect, but easier to read.